status_history.go

```diff
@@ -0,0 +1,21 @@
+package braintree
+
+import (
+	"time"
+)
+
+// StatusHistory contains information about the last 50 timestamps where
+// something changed about the subscription.
+type StatusHistory struct {
+	StatusEvents []StatusEvent `xml:"status-event"`
+}
+
+// StatusEvent contains information about what and when something
+// changed about the subscription.
+type StatusEvent struct {
+	Timestamp          time.Time `xml:"timestamp"`
+	Balance            *Decimal  `xml:"balance"`
+	Price              *Decimal  `xml:"price"`
+	Status             string    `xml:"status"`
+	SubscriptionSource string    `xml:"subscription-source"`
```

Comment:
Could you also include the `currency-iso-code`, `user`, and `plan-id` fields?

Reply:
"plan-id" is not part of the status events but of the Transaction itself:

// TODO: not all transaction fields are implemented yet, here are the missing fields (add on demand)
//
// <transaction>
//   <voice-referral-number nil="true"></voice-referral-number>
//   <status-history type="array">
//     <status-event>
//       <timestamp type="datetime">2013-10-07T17:26:14Z</timestamp>
//       <status>authorized</status>
//       <amount>7.00</amount>
//       <user>eaigner</user>
//       <transaction-source>Recurring</transaction-source>
//     </status-event>
//     <status-event>
//       <timestamp type="datetime">2013-10-07T17:26:14Z</timestamp>
//       <status>submitted_for_settlement</status>
//       <amount>7.00</amount>
//       <user>eaigner</user>
//       <transaction-source>Recurring</transaction-source>
//     </status-event>
//     <status-event>
//       <timestamp type="datetime">2013-10-08T07:06:38Z</timestamp>
//       <status>settled</status>
//       <amount>7.00</amount>
//       <user nil="true"></user>
//       <transaction-source></transaction-source>
//     </status-event>
//   </status-history>
//   <plan-id>bronze</plan-id>
//   <subscription-id>jqsydb</subscription-id>
//   <subscription>
//     <billing-period-end-date type="date">2013-11-06</billing-period-end-date>
//     <billing-period-start-date type="date">2013-10-07</billing-period-start-date>
//   </subscription>
//   <add-ons type="array"/>
//   <discounts type="array"/>
//   <descriptor>
//     <name nil="true"></name>
//     <phone nil="true"></phone>
//   </descriptor>
//   <recurring type="boolean">true</recurring>
// </transaction>

from transaction.go

The rest of the fields I will add.

Reply:
Argh, never mind. I looked at the Transaction history, not the Subscription history. I will add the plan-id field as well.